Bloom.md

```diff
@@ -0,0 +1,341 @@
+
+We evaluated the following libraries:
+* https://crates.io/crates/bloomfilter
+* https://crates.io/crates/growable-bloom-filter
+* https://crates.io/crates/fastbloom
+
+Here are some important factors when evaluating libraries:
+* Is it actively maintained?
+* Has there been any breaking changes in serialization and deserialization of bloom filters over version updates?
+* Performance
+* Popularity
+
+https://crates.io/crates/bloomfilter was chosen as it provides all the required APIs for controlling a bloom filter.
+This library has also been stable through the releases.
+
+Certain libraries (e.g. fastbloom) are no longer compatible with older versions (after updates) resuling in
+serialization / deserialization incompatibility.
+
+growable-bloom-filter is a bloom filter library that implements a scalable bloom filter. However, it does not allow
+control over when the scaling occurs and does not expose APIs to check the number of filters that it has scaled out to.
+This crate also has had a breaking change across versions and bloom filters of the older version are not loadable on the
+newer versions.
+
+
+### Large BloomFilter objects
+
+Create and Delete operations on large bloom filters take longer durations and will block the main thread for this duration. 
+Because of this, the following operations will be handled differently:
+* defrag callback: If the memory used by any bloom filter within the bloom object is greater than X MB (`bloom_large_item_threshold`
+    constant), we will skip defrag operations on this bloom object. Otherwise, we will defrag the bloom object in iterations
+    for each bloom filter in the bloom object.
+* free_effort callback: This callback decides the free effort for the bloom object. If it is greater than X MB
+    (`bloom_large_item_threshold` constant), we will return 0 to use async free on the bloom object.
+* create operations (BF.ADD/MADD/INSERT/RESERVE): We can compute the estimated size of the bloom filter that will be created as
+    a result of the operation and we can consider rejecting requests in case of unavailable memory. Scalable Bloom filters
+    will grow in used memory after creation - but only as a result of a BF.ADD, BF.MADD, or BF.INSERT operation and we can reject
+    these requests if we identify that it will result in a scale out operation that requires more memory than what is available.
+
+
+## Specification
+
+### Bloom Filter Command API
+
+The following Bloom Filter commands are supported and their API syntax is compatible with ReBloom:
+BF.ADD <key> <item>
+BF.EXISTS <key> <item>
+BF.MADD <key> <item> [<item> ...]
+BF.MEXISTS <key> <item> [<item> ...]
+BF.CARD <key>
+BF.INFO <key> [CAPACITY | SIZE | FILTERS | ITEMS | EXPANSION]
+BF.RESERVE <key> <false_positive_rate> <capacity> [EXPANSION <expansion>] | [NONSCALING]
+BF.INSERT <key> [CAPACITY <capacity>] [ERROR <fp_error>] [EXPANSION <expansion>] [NOCREATE] [NONSCALING] ITEMS <item> [<item> ...]
```

Comment:
Would prefer request / response for each command in the specification.

Reply:
+1 also, why not BF.SET. We should follow the naming conventions that valkey has today. BF.SET can probably handle both cases, creation of a new filter and adding items.

Reply:
I have updated the section to list all the APIs and also list the responses.